ui/src/features/auth/oidc-login.tsx

```diff
@@ -44,9 +56,10 @@ export const OIDCLogin = ({ oidcConfig }: Props) => {
     queryKey: [issuerUrl],
     queryFn: () =>
       issuerUrl &&
-      oauth
-        .discoveryRequest(issuerUrl)
-        .then((response) => oauth.processDiscoveryResponse(issuerUrl, response))
+      discoveryRequest(issuerUrl, {
+        [allowInsecureRequests]: shouldAllowHttpRequest()
```

Comment:
@krancour FYI this will block auth if IDP is not configured with https. I want to make sure there is absolutely no requirement of http request to IDP except it is local development environment.